From daffe65aa5848f5ea63593447c6e73f47ceeaaa9 Mon Sep 17 00:00:00 2001
From: Devanshi Bansal <quic_devanshi@quicinc.com>
Date: Thu, 28 Aug 2025 15:31:15 +0530
Subject: [PATCH 1/3] QCOM: libgbm: create MSM backend if KGSL GPU

Add support for the MSM backend in libgbm to enable compatibility with
Qualcomm Adreno GPUs using the KGSL driver. This change allows libgbm
to detect and utilize the MSM backend when a KGSL GPU is present,
enabling vendor-specific graphics features on Qualcomm platforms.

Change-Id: I0f42c11a24a21ade4f046059b601d921d600c74d
Signed-off-by: Devanshi Bansal <quic_devanshi@quicinc.com>
Upstream-Status: Pending
---
 src/gbm_msm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gbm_msm.c b/src/gbm_msm.c
index f7e4bae..cd05fd3 100644
--- a/src/gbm_msm.c
+++ b/src/gbm_msm.c
@@ -719,6 +719,11 @@ gbm_msm_surface_create(struct gbm_device *gbm,
 static struct gbm_device *
 msm_device_create(int fd, uint32_t gbm_backend_version)
 {
+   const char *gpu_path = "/sys/class/kgsl/kgsl-3d0/";
+   if (access(gpu_path, F_OK) != 0) {
+      return NULL;
+   }
+
    struct gbm_msm_device *msm;
 
    assert(gbm_core_->v0.core_version == GBM_BACKEND_ABI_VERSION);
-- 
2.34.1

